consensus/ssz/src/decode/try_from_iter.rs

```diff
@@ -29,11 +29,18 @@ pub trait TryFromIter<T>: Sized {
 impl<T> TryFromIter<T> for Vec<T> {
     type Error = Infallible;
 
-    fn try_from_iter<I>(iter: I) -> Result<Self, Self::Error>
+    fn try_from_iter<I>(values: I) -> Result<Self, Self::Error>
     where
         I: IntoIterator<Item = T>,
     {
-        Ok(Self::from_iter(iter))
+        // Pre-allocate the max size of the Vec. We expect to receive vastly more valid
```

Comment:
I'm a little confused -- is this meant to pre-allocate the `max_size` of the variable list or `num_items` in the list? It looks like this allocates `num_items` based on the range you linked:

https://github.com/sigp/lighthouse/blob/2983235650811437b44199f9c94e517e948a1e9b/consensus/ssz/src/decode/impls.rs#L489

But the comments and variable names make me think the intent might be to allocate the max length of the variable list?

Regardless the change here does look equivalent to what we were doing prior to #3371

Reply:
Oh yeah, allocating `num_items` is the intention. I've updated the comment to hopefully be a bit clearer about this

Reply:
ok this looks good to me then!